tests/tensor_search/test_model_cache_management.py

```diff
@@ -0,0 +1,146 @@
+        if self.CUDA_FLAG:
+            assert len(available_models) >= 6
+        else:
+            assert len(available_models) >= 3
+
+    def tearDown(self) -> None:
+        clear_loaded_models()
+
+    def test_eject_model_cpu(self):
+        for model_name in self.MODEL_LIST:
+            eject_model(model_name, "cpu")
+            if (model_name, "cpu") in available_models:
+                raise AssertionError
+
+
+        my_test_model_1 = "test-model-1"
+        my_test_model_2 = "test-model-2"
+
+        try:
+            eject_model(my_test_model_1, "cpu")
+        except ModelNotInCache:
+            pass
+
+        try:
+            eject_model(my_test_model_2, "cpu")
+        except ModelNotInCache:
+            pass
+
+
+    def test_eject_model_cuda(self):
+        if self.CUDA_FLAG:
+        # check if we can eject the models
+            for model_name in self.MODEL_LIST:
+                eject_model(model_name,"cuda")
+                if (model_name, "cuda") in available_models:
+                    raise AssertionError
+            my_test_model_1 = "test-model-1"
+            my_test_model_2 = "test-model-2"
+
+            try:
+                eject_model(my_test_model_1, "cuda")
+            except ModelNotInCache:
+                pass
+
+            try:
+                eject_model(my_test_model_2, "cuda")
+            except ModelNotInCache:
+                pass
+        else:
+            pass
+
+
+    def test_cuda_info(self):
+        try:
+            get_cuda_info()
+        except HardwareCompatabilityError:
+            pass
+
+
+    def test_loaded_models(self):
+
+        loaded_models = get_loaded_models()["models"]
+        loaded_models_keys = [_create_model_cache_key(dic["model_name"], dic["model_device"],
+                                _validate_model_properties(dic["model_name"], None)) for dic in loaded_models]
+        assert loaded_models_keys==list(available_models.keys())
+
+
+    def test_edge_case_cuda(self):
+        if self.CUDA_FLAG:
+            test_iterations = 10
+            # Note this is a time consuming test.
+
+            for i in range(test_iterations):
+                eject_model(self.MODEL_1, "cuda")
+                load_model(self.MODEL_1, "cuda")
+
+                for id in range(torch.cuda.device_count()):
+                    # cuda usage
+                    assert torch.cuda.memory_allocated(id) < torch.cuda.get_device_properties(id).total_memory
+                # cpu usage
+                assert psutil.cpu_percent(1) < 100.0
+                # memory usage
+                assert psutil.virtual_memory()[2]< 100.0
+        else:
+            pass
+
+
+    def test_edge_case_cpu(self):
+        test_iterations = 10
+        # Note this is a time consuming test.
+
+        for i in range(test_iterations):
+            eject_model(self.MODEL_1, "cpu")
+            load_model(self.MODEL_1, "cpu")
+            # cpu usage
```

Comment:
Can we check that the model is present when loaded, and not present when ejected?

Reply:
added a new test